integration_tests/integration/serviceProviderReferrals.cy.js

```diff
@@ -1556,13 +1554,133 @@ describe('Service provider referrals dashboard', () => {
+          appointmentDeliveryType: 'PHONE_CALL',
+        }),
+        actionPlanAppointmentFactory.attended('no').build({
+          sessionNumber: 2,
+          appointmentTime: '2021-05-31T09:02:02Z',
+          durationInMinutes: 75,
+          appointmentDeliveryType: 'PHONE_CALL',
+        }),
+        actionPlanAppointmentFactory.build({
+          sessionNumber: 3,
+          appointmentTime: '2021-07-31T09:02:02Z',
+          durationInMinutes: 75,
+          appointmentDeliveryType: 'PHONE_CALL',
+        }),
+      ]
+
+      const assignedReferral = sentReferralFactory.assigned().build({
+        ...referralParams,
+        assignedTo: { username: serviceProvider.username },
+        actionPlanId: actionPlan.id,
+      })
+
+      cy.stubGetSentReferralsForUserTokenPaged(pageFactory.pageContent([]).build())
+      cy.stubGetServiceCategory(serviceCategory.id, serviceCategory)
+      cy.stubGetIntervention(intervention.id, intervention)
+      cy.stubGetSentReferral(assignedReferral.id, assignedReferral)
+      cy.stubGetServiceUserByCRN(assignedReferral.referral.serviceUser.crn, deliusServiceUser)
+      cy.stubGetUserByUsername(probationPractitioner.username, probationPractitioner)
+      cy.stubGetSupplierAssessment(assignedReferral.id, supplierAssessmentFactory.build())
+      cy.stubGetAuthUserByUsername(serviceProvider.username, serviceProvider)
+
+      cy.stubGetActionPlan(actionPlan.id, actionPlan)
+      cy.stubGetApprovedActionPlanSummaries(assignedReferral.id, [
+        { id: actionPlan.id, submittedAt: actionPlan.submittedAt, approvedAt: actionPlan.approvedAt },
+      ])
+      cy.stubGetActionPlanAppointments(actionPlan.id, appointments)
+
+      cy.login()
+      cy.visit(`/service-provider/referrals/${assignedReferral.id}/progress`)
+      cy.get('[data-cy=session-table]')
+        .getTable()
+        .should('deep.equal', [
+          {
+            'Session details': 'Session 1',
+            'Date and time': '9:02am on 24 Mar 2021',
+            Status: 'scheduled',
+            Action: 'Reschedule session\n\t\t\t\t\t\t\n\t\t\t\t\t\t\t\tGive feedback',
+          },
+          {
+            'Session details':
+              'Previous appointments\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\n\t\t\t\n\n\t\t\t\t\t\t\n \t\t\t\t\t\t\t \n\t\t\n\n\t\t\t\t\t\t\n\t\t\t\t\t\t\t\n\t\t\t\t\t\t \t\t\n\t\t\t\t\t\t\t\t\tSession 1 \n\t\t\t\t\t\t\t\t\t10:02am on 23 Apr 2021\n\t\t\t\t\t\t\t\t\t\n  did not attend\n\n\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\t\tView feedback form\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\n\n\t\t\t\t\t \t\t\n \t\t\t\t\t\t\t \n\t\t\n\n\t\t\t\t\t\t\n\t\t\t\t\t\t\t\n\t\t\t\t\t\t \t\t\n\t\t\t\t\t\t\t\t\tSession 1 \n\t\t\t\t\t\t\t\t\t10:02am on 23 May 2021\n\t\t\t\t\t\t\t\t\t\n  did not attend\n\n\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\t\tView feedback form',
```

Comment:
(reposting here in case slack message was missed):
this output
```
          {
            'Session details':
              'Previous appointments\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\n\t\t\t\n\n\t\t\t\t\t\t\n \t\t\t\t\t\t\t \n\t\t\n\n\t\t\t\t\t\t\n\t\t\t\t\t\t\t\n\t\t\t\t\t\t \t\t\n\t\t\t\t\t\t\t\t\tSession 1 \n\t\t\t\t\t\t\t\t\t10:02am on 23 Apr 2021\n\t\t\t\t\t\t\t\t\t\n  did not attend\n\n\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\t\tView feedback form\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\n\n\t\t\t\t\t \t\t\n \t\t\t\t\t\t\t \n\t\t\n\n\t\t\t\t\t\t\n\t\t\t\t\t\t\t\n\t\t\t\t\t\t \t\t\n\t\t\t\t\t\t\t\t\tSession 1 \n\t\t\t\t\t\t\t\t\t10:02am on 23 May 2021\n\t\t\t\t\t\t\t\t\t\n  did not attend\n\n\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t\t\t\t\t\t\t\tView feedback form',
          },
```
actually looks like a bug in cypress - the control characters seem to broken something when its trying to stringify the value


if you change that one to  something like
```js
        .should(result => {
          expect(result).to.have.length(6)
          expect(result[0]).to.deep.include({
            'Session details': 'Session 1',
            'Date and time': '9:02am on 24 Mar 2021',
            Status: 'scheduled',
          })
          expect(result[0].Action).to.match(/^Reschedule session[\n|\t]*Give feedback$/)
          expect(result[1]).to.deep.include({
            'Session details': 'Session 1',
            'Date and time': '10:02am on 23 Apr 2021',
            Status: 'did not attend',
            Action: 'View feedback form',
          })
          // + check the rest of the session rows 
        })
 ```
it'll be clearer anyway